### ISSUE TITLE
Tag assignment bugs

### ISSUE THREAD

**Issue by phraenquex** [m2ms/fragalysis-frontend#789]
1. * "Assign tags" needs a scroll bar
2. * "Assign tags" needs a "not all selected" symbol (asterisk?)
3. * show how many hits are selected (next to "Unselect" button
4. * Untagging with a partially-tagging tag (e.g. "Active site" (Mpro)) sometimes doesn't seem to do anything (screenshot below)

For point 4 above: do this in Mpro:  clear the hit navigator, then filter "S1-prime extensions", then click the edit pencil, then click on "Active site" (which is already highlighted) - but nothing happens.  Playing a bit more (using the "Intersection" option), you find that tagging with "Active site" tags (or untags) them one hit at a time.  Definitely broken.
![image](https://user-images.githubusercontent.com/33267937/152773789-35af0f89-c93b-474d-8d4a-70015ff2c041.png)

Another example: in Mpro, 
* select "Active site", and "isoquinoline" and "Intersection" (puts 13 in hit navigator).  
* Unselect "Active site", then "select all hits", then try to tag "Active site" - then select "Active site" again *(puts some unpredictable number in hit navigator)*

**Comment by rsanchezgarc** [comment]
May I suggest having some "protected" tags that only admin users can modify?

**Comment by boriskovar-m2ms** [comment]
One of the problems is that when you untag some hits in Tag assignment window then these molecules disappear from the hit navigator. Previously I implemented functionality that tagging and untagging hit in Tag editor (now what is actually Tag details but was in place of Tag assignment window) wouldn't remove them from hit navigator and given hits would be removed only after the editing window (previously tag editor and now tag assignment) was closed. This can cause a lot of confusion.

**Comment by boriskovar-m2ms** [comment]
@duncanpeacock  The source of (at least some) of the glitches is that I can only assign (or remove) tag from hit one by one i.e. if I have 400 selected hits in the hit navigator and assign tag to them, it means that I must perform 400 api calls. I think it's needed that we have a api batch update.

**Comment by phraenquex** [comment]
I confirm that this bug is a serious limitation to tag usage;  but what is critical is reliability; speed is NOT critical.

@duncanpeacock please estimate the work involved in a batch-wise tagging, for discussion later today.  Create a new ticket if necessary.
@boriskovar-m2ms also suggest hacky work-arounds, e.g. pop up a modal while you send out requests with 0.1s delays between API calls.

**Comment by rsanchezgarc** [comment]
@boriskovar-m2ms I cannot get the same numbers that Frank gets. My intersection between Active site and Isoquinoline contains only one molecule
